src/metabase/lib/convert.cljc

```diff
@@ -491,3 +497,26 @@
       (throw (ex-info (lib.util/format "Error converting MLv2 query to legacy query: %s" (ex-message e))
                       {:query query}
                       e)))))
+
+(mu/defn legacy-ref->pMBQL :- ::lib.schema.ref/ref
```

Comment:
Added a helper function for converting legacy MBQL refs to pMBQL and doing JS->Cljs conversion and normalization since we needed it in several places.